src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_BinaryOperator.cs

```diff
@@ -111,6 +111,31 @@ public override BoundNode VisitUserDefinedConditionalLogicalOperator(BoundUserDe
 
         public BoundExpression VisitBinaryOperator(BoundBinaryOperator node, BoundUnaryOperator? applyParentUnaryOperator)
         {
+            if (node.InterpolatedStringHandlerData is InterpolatedStringHandlerData data)
+            {
```

Comment:
Consider extracting the body of this if into its own function to make it clearer that we're doing the lowering in a different way.